common/src/main/java/io/netty/util/concurrent/SingleThreadEventExecutor.java

```diff
@@ -355,52 +378,92 @@ protected boolean removeTask(Runnable task) {
      * @return {@code true} if and only if at least one task was run
      */
     protected boolean runAllTasks() {
+        assert inEventLoop();
         boolean fetchedAll;
+        boolean ranAtLeastOne = false;
+
         do {
             fetchedAll = fetchFromScheduledTaskQueue();
-            Runnable task = pollTask();
-            if (task == null) {
-                return false;
+            if (runAllTasksFrom(taskQueue)) {
+                ranAtLeastOne = true;
             }
+        } while (!fetchedAll); // keep on processing until we fetched all scheduled tasks.
 
-            for (;;) {
-                try {
-                    task.run();
-                } catch (Throwable t) {
-                    logger.warn("A task raised an exception.", t);
-                }
+        if (ranAtLeastOne) {
```

Comment:
nvm ... different overload of `runAllTasksFrom` is being used in this case which doesn't use `setLastExecutionTime`.